pysr/regressor_sequence.py

```diff
@@ -0,0 +1,320 @@
+            weights,
+            variable_names,
+            X_units,
+        )
+        self.variable_names = variable_names  # for latex_table()
+        self.n_features = X.shape[1]  # for latex_table()
+
+        current_X = X[self.recursive_history_length :]
+        historical_X = self._sliding_window(X)[: -1 : current_X.shape[1], :]
+        y_units = X_units
+        if isinstance(weights, np.ndarray):
+            weights = weights[self.recursive_history_length :]
+        variable_names, display_variable_names = self._construct_variable_names(
+            current_X.shape[1], variable_names
+        )
+
+        self._regressor.fit(
+            X=historical_X,
+            y=current_X,
+            weights=weights,
+            variable_names=variable_names,
+            display_variable_names=display_variable_names,
+            X_units=X_units,
+            y_units=y_units,
+            complexity_of_variables=complexity_of_variables,
+        )
+        return self
+
+    def predict(self, X, index=None, num_predictions=1):
+        """
+        Predict future data from input X using the equation chosen by `model_selection`.
+
+        You may see what equation is used by printing this object. X should
+        have the same columns as the training data.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Data of shape `(n_times, n_features)`.
+        index : int | list[int]
+            If you want to compute the output of an expression using a
+            particular row of `self.equations_`, you may specify the index here.
+            For multiple output equations, you must pass a list of indices
+            in the same order.
+        num_predictions : int
+            How many predictions to make. If `num_predictions` is less than
+            `(n_times - recursive_history_length + 1)`,
+            some input data at the end will be ignored.
+            Default is `1`.
+
+        Returns
+        -------
+        x_predicted : ndarray of shape (num_predictions, n_features)
+            Values predicted by substituting `X` into the fitted sequence symbolic
+            regression model and rolling it out for `num_predictions` steps.
+
+        Raises
+        ------
+        ValueError
+            Raises if the `best_equation` cannot be evaluated.
+        """
+        X = self._validate_data(X, ensure_2d=False)
+        if X.ndim == 1:
+            X = X.reshape(-1, 1)
+        assert X.ndim == 2
+        _check_assertions(X, recursive_history_length=self.recursive_history_length)
+        historical_X = self._sliding_window(X)[:: X.shape[1], :]
+        if num_predictions < 1:
+            raise ValueError("num_predictions must be greater than 0.")
+        if num_predictions < len(historical_X):
+            warnings.warn(
+                "The number of predictions is less than the number of historical data points. Some will be ignored."
```

Comment:
What is being ignored here? Sorry I'm not sure I understand this branch of the code.

Reply:
so lets say the user inputted `[1, 2, 3, 4, 5]` and `num_predictions = 1`
then historical_X would be
```
[
[1, 2, 3],
[2, 3, 4],
[3, 4, 5]
]
```
since num_predictions is 1, `[2, 3, 4]` and `[3, 4, 5]` are ignored here and are not used to predict anything

Reply:
Got it! Thanks. In this case I don't think we need the warning. It could also be true that some variables are ignored in the equation itself and we don't warn for that either